daffodil-core/src/main/scala/org/apache/daffodil/dsom/SimpleTypes.scala

```diff
@@ -53,9 +54,9 @@ trait TypeBase {
 trait NonPrimTypeMixin
 
 sealed trait SimpleTypeBase extends TypeBase
-  with HasOptRepTypeMixin {
+  with HasOptRepTypeMixin with SimpleTypeView {
 
-  def primType: PrimType
+  override def primType: PrimType
```

Comment:
Giving direct access to PrimType is potentially problematic from a backwards compatibility perspective. This is something that has changed fairly frequently. I wonder if we also want a walker specific view of PrimType's as well?

Reply:
I can understand how that could cause some issues.  What would be a more stable way to access information about a simple type without exposing the internal classes?  Would it be enough to simply add a String member of SimpleTypeBase that returns `primType.toString`?  My NiFi Record implementation is very dependent on pattern matching Simple types, and I imagine most implementations would be as well.

Reply:
You could probably just use the same trait trick same as the Views (e.g.  trait that's like org.apache.daffodil.dsom.walker.IntegerView which extends dsom.walker.PrimTypeView) and then the dpath.PrimType objects extend the appropriate view. Maybe this is excessive though? I'm trying to imagine a clean API where the user only has access to things in dsom.walker.*  There's a lot of duplication though.

Reply:
You mentioned that `PrimType` is changed frequently - would this list pose an issue because of that? i.e. wouldn't future developers have to go change this list of primitive Views every time `PrimType` is changed?

Reply:
I think list of PrimTypes isn't changed that often (and likely won't change anytime soon). The things that do change relatively frequently is the functions defined in the prim types. Because these change so often, we might not want those members to be publicly visible to a dsom walker. The View trait allows us to decide which members are made visible, which I suspect is not going to be many, if any at all. So you can still match on thes prim type views to figure out the type.

You are correct that if a new PrimType is added, we'll also need to add a PrimTypeView as well. Not much we can do about that though.

Reply:
I went ahead and defined an Enum type called PrimTypeView and created Enum representations for all the "PrimType" enum types within it.  I could not find a way to make the NodeInfo class import *anything* from `dsom.walker`, so I was unable to make it extend any trait I defined there.  Instead, I added a new field to `SimpleTypeBase` called `primTypeView` which just matches `primType` to my new Enum type.  Let me know if you think this approach should be changed; the pattern matching is done in `SimpleTypeBase` itself.

Reply:
Doesn't seem unreasonable. I had forgotten that NodeInfo is defined in daffodil-runtime1. The problem is daffodil-core depends on daffodil-runtime1, (core creates many of the objects defined in runtime1), so runtime1 can't use anything defined in core or we'd have a circular dependency. They way you handled it seems reasonable.

If you wanted to do it with mixin views, you would need to define the PrimTypeView traits in the daffodil-runtime1 class. They can still be in a dsom.walker package, just in a different subproject/jar.

Reply:
I went ahead with that route, it seemed a lot more elegant to integrate the View paradigm without adding any fields, just like what had been done for everything else.  I put all the Views related to PrimType in their own file in runtime-1 and most of the primitive enums now implement them